dev/tests/integration/testsuite/Magento/Catalog/_files/product_special_price_rollback.php

```diff
@@ -4,17 +4,26 @@
  * See COPYING.txt for license details.
  */
 
+use Magento\Catalog\Model\ProductRepository;
+use Magento\CatalogInventory\Model\StockRegistryStorage;
+
 /** @var \Magento\Framework\Registry $registry */
 $registry = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->get(\Magento\Framework\Registry::class);
 
 $registry->unregister('isSecureArea');
 $registry->register('isSecureArea', true);
 
-$repository = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
-    \Magento\Catalog\Model\ProductRepository::class
-);
+$productSku = 'simple';
+$objectManager = \Magento\TestFramework\Helper\Bootstrap::getObjectManager();
+/** @var ProductRepository $repository */
+$repository = $objectManager->get(ProductRepository::class);
+/** @var StockRegistryStorage $stockRegistryStorage */
+$stockRegistryStorage = $objectManager->get(StockRegistryStorage::class);
+
 try {
-    $product = $repository->get('simple');
+    $product = $repository->get($productSku);
+    // remove stock item from cache
+    $stockRegistryStorage->removeStockItem($product->getId());
```

Comment:
Why do we need it? Problem is that Stock is cached for productId. The situation is next: 
We have test Magento\Catalog\Pricing\Render\FinalPriceBox::testRenderingAccordingToIsProductListFlag() which has 2 data sets. When test is running with first data set in setUp() we are trying to get product with `$productRepository->get('simple');` When we try to get product it will try to get stock item for this product and cache it in `Magento\CatalogInventory\Model\StockRegistryStorage::setStockItem()`. So Stock Item is cached. When test is running with second data set and again use fixture Magento/Catalog/_files/product_special_price.php it will get Stock Item from cache and try to save it again. As Stock Item in cache has id, so it will try to update it, but we don't have stock item with such id in db. So nothing will happen and product will be without Stock Item in db. When we use Magento\Inventory\Model\SourceItem\Validator\StatusConsistencyValidator::validate() it will try to get Stock Item from db, but it will find nothing so it set `$isManageStock = true`, but product should has `"manage_stock" = 0` and product doesn't have qty, so it leads to exception and test fails.

Reply:
You can remove it

Reply:
Test will fail.